File: python_tests/test_2_even_numbers.py
# Task 2
# Write a program to display the list of even numbers.

# Implementation
def get_even_numbers(n: int) -> []:
    even_number_list = filter(lambda x: x % 2 == 0, range(1, n))
    return list(even_number_list)


# Test Section
def test_even_numbers_10():
    n = 10
    expected_list = [2, 4, 6, 8]
    assert get_even_numbers(n) == expected_list


def test_even_numbers_20():
    n = 20
    expected_list = [2, 4, 6, 8, 10, 12, 14, 16, 18]
    assert get_even_numbers(n) == expected_list

File: CodingInterview/CodingInterview/csharp_tests/test_2_even_numbers.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Linq;
using NUnit.Framework;
//Task 2
// Write a program to display the list of even numbers.
namespace CodingInterview
{
    public class Class2
    {

        private static List<int> GetEvenNumbers(int n)
        {
            var asList = Enumerable
                .Range(1, n)
                .ToList()
                .FindAll(e => e % 2 == 0)
                .ToList();
            asList.ForEach(i => Console.Write("{0}\t", i));
            return asList;
        }

        [Test]
        public void CheckEvenNumbers()
        {
            var ExpectedEvenList = new List<int>() {2,4,6,8};
            var n = 9;
            Assert.AreEqual(GetEvenNumbers(n), ExpectedEvenList);
        }
    }

}